tardis/transport/montecarlo/single_packet_loop.py

```diff
@@ -85,6 +85,9 @@ def single_packet_loop(
     )
 
     rpacket_tracker.track(r_packet)
+    rpacket_tracker.track_boundary_interaction(
```

Comment:
just do track_boundary after the event has happened and do current_shell_id - delta_shell, current_shell_id.

Reply:
Updated the code.

Reply:
The output
![image](https://github.com/user-attachments/assets/e11c81b0-68d3-4f0f-9758-7bf431b1d81b)
![image](https://github.com/user-attachments/assets/b0541d6c-88e4-4aff-80a2-a7ae10b0a2e5)